janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrResultIterator.java

```diff
@@ -63,7 +63,7 @@ public SolrResultIterator(SolrClient solrClient, Integer limit, int offset, int
 
     @Override
     public boolean hasNext() {
-        if (count != 0 && count % batchSize == 0 && batchSize * numBatches < limit) {
+        if (count != 0 && count % batchSize == 0 && count < limit) {
```

Comment:
This change _seems_ reasonable, but was this a Coverity-specified error or a separate change? I had to convince myself that this is correct, but this is a behavior change, so I'm curious how none of the unit tests failed before with this, and none required to be changed given this change (i.e., it should be possible to reproduce this error in a unittest such that it fails before, and succeeds after this change; or, that it succeeds before with an erroneous assertion, and this change requires changing said assertion to be correct).

Should this be a separate change?

Reply:
@mbrukman The Coverity finding was that `batchSize*numBatches` is performed at int32 but limit is int64. Upon review though I realized when `count % batchSize == 0` (e.g. at the start of a new batch), `count` is equal to `batchSize * numBatches`. I was familiar with this line from reviewing when it was added (see https://github.com/JanusGraph/janusgraph/pull/481#discussion_r136208445). @davidclement90 Does this seem okay?

Reply:
Agreed, LGTM.

Reply:
Yes it's ok LGTM